src/Simulation/Simulators.Tests/TestProjects/ClassicControlSupportTest/Main.qs

```diff
@@ -0,0 +1,13 @@
+﻿namespace Quantum.ClassicControlSupportTest {
+    open Microsoft.Quantum.Intrinsic;
+    open Microsoft.Quantum.Simulation.QuantumProcessor.Extensions;
+    @EntryPoint()
+    operation Main () : Unit {
+        using (q = Qubit())
+        {
+            H(q);
+            let result = M(q);
+            ApplyIfOne(result, (X, q));

```

Comment:
Instead of just a single test with an explicit call to ApplyIfOne, you should instead target the ClassicControlSupportTest.csproj project to the Honeywell processor, and test all the cases that you also test on the compiler ([here](https://github.com/microsoft/qsharp-compiler/blob/master/src/QsCompiler/Tests.Compiler/TestCases/ClassicalControl.qs)).

Reply:
Am I allowed to put something specific to Honeywell in this repo? Are we allowed to even talk about that here on GitHub?

Reply:
Yes. :)

Reply:
Ah okay, I thought everything related to specific partners was confidential, but it seems that is in the past now.

Reply:
I've updated the tests to reference each of the API operations. I don't think targeting a specific processor is going to have an effect here as the API referenced directly.

Reply:
That still doesn't test whether a project that has been compiled for Honeywell will actually execute on the QuantumSimulator; if e.g. the compiler changes what operations it relies on when translating if-statements, these tests will still pass fine.

Reply:
Hmm... Scott and I had discussed the possibility of doing a more general test of the conditional rewrite logic and thought that might be out of scope for this change, but it sounds like that was more along the lines of what you were imagining here. It felt a little awkward to use Honeywell specifically for testing this. Is there any merit in creating a test target that just invokes all rewrite steps without being specific to a partner or has any connections to submission logic? That was the other idea we discussed, and while that addition would have to go into the compiler repo first it would give us a nice way to invoke everything needed for testing without relying on a specific partner target.

Reply:
Some of the steps may be mutually exclusive. 
What we want to test is always the functionality, and not the implementation - and certainly not the implementation in another repository...
The functionality here we want to provide that should be tested is that for all execution target, the built executable will execute on the QuantumSimulator and the Resources estimator as well. Testing that exhaustively is a slightly ambitious endeavor (there are general frameworks for other languages to generate "sample" programs for similar purposes, but I haven't looked into what sort of tooling exist for us). Hence my suggestion is to for now to be a bit less ambitious and test a handful of programs that we know will be modified during compilation and test those for all execution targets. Setting up a project compiled for Honeywell, one compiled for QCI and on for IonQ actually seem like to right thing to do to me. As a compromise to keep the testing effort reasonable (though strictly speaking that also deviates from testing exactly the functionality we want) and not literally generate a separate compilation unit for each program and target to test, you could leverage the NoEntryPoint property we've introduced specifically for testing purposes. The other option would be to do an in-memory compilation, but that would bypass the Sdk part of the setup then, so also not fully reflect the functionality.

Reply:
As a side note: The reason that the test for the QsharpExe actually does an out of process invocation is because we specifically wanted to test the command line functionality. For the tests here, it would be fine to go with Q# unit tests (making the above suggested compromise of relying on the guarantee of NoEntryPoint that all code will still be fully targeted).